src/games/strategy/triplea/attatchments/UnitAttachment.java

```diff
@@ -799,30 +798,16 @@ public void resetUnitPlacementRestrictions() {
     m_unitPlacementRestrictions = null;
   }
 
-  // no m_ variable for this, since it is the inverse of m_unitPlacementRestrictions we might as well just use
-  // m_unitPlacementRestrictions
+  // no m_ variable for this, since it is the inverse of m_unitPlacementRestrictions
+  // we might as well just use m_unitPlacementRestrictions
   @GameProperty(xmlProperty = true, gameProperty = true, adds = false)
   public void setUnitPlacementOnlyAllowedIn(final String value) throws GameParseException {
-    StringBuilder valueRestricted = new StringBuilder();
-    final String valueAllowed[] = value.split(":");
-    if (valueAllowed != null) {
-      getListedTerritories(valueAllowed);
-      for (final Territory item : getData().getMap().getTerritories()) {
-        boolean match = false;
-        for (final String allowed : valueAllowed) {
-          if (allowed.equals(item.getName())) {
-            match = true;
-            break;
-          }
-        }
-        if (!match) {
-          valueRestricted.append(":").append(item.getName());
-        }
-      }
-
-      String newValue = valueRestricted.toString().replaceFirst(":", "");
-      m_unitPlacementRestrictions = newValue.split(":");
-    }
+    final Collection<Territory> allowedTerritories = getListedTerritories(value.split(":"));
+    final Collection<Territory> restrictedTerritories = new HashSet<>(getData().getMap().getTerritories());
+    restrictedTerritories.removeAll(allowedTerritories);
+    m_unitPlacementRestrictions = restrictedTerritories.stream()
+        .map(Territory::getName)
+        .toArray(size -> new String[size]);
```

Comment:
Mostly out of curiosity, do you need to set the array size here? I haven't used stream much yet so just wondering.